Add mark-as-unread

Alkarex opened the pull request:
https://github.com/FreshRSS/FreshRSS/issues/1966

Alkarex committed:
Add mark-as-unread

https://github.com/FreshRSS/FreshRSS/issues/1966

Frenzie commented on app/i18n/en/feedback.php:
```diff
@@ -52,6 +52,10 @@
 	'sub' => array(
 		'actualize' => 'Updating',
+		'articles' => array(
+			'marked_read' => 'The articles have been marked as read.',
```
The selected articles have been marked as read?

Frenzie commented on app/i18n/nl/index.php:
```diff
@@ -40,6 +40,7 @@
 		'mark_all_read' => 'Markeer alles als gelezen',
 		'mark_cat_read' => 'Markeer categorie als gelezen',
 		'mark_feed_read' => 'Markeer feed als gelezen',
+		'mark_selection_unread' => 'Mark selection as unread',	//TODO
```
Markeer selectie als ongelezen

Frenzie replied:
Alright, I'll wait until several of these are merged (or at the very least until tomorrow) before I do that. :-)

Frenzie reviewed (approved):
I'll toss in a follow-up PR for the Dutch translation if you prefer. :-)

Frenzie commented on app/layout/nav_menu.phtml:
```diff
@@ -77,8 +78,12 @@
 				'search' => htmlspecialchars_decode(FreshRSS_Context::$search, ENT_QUOTES),
 				'state' => FreshRSS_Context::$state,
-			)
+			),
```
I always get annoyed by languages that don't annoy this. ;-)

Alkarex replied:
Like JSON!

Alkarex committed:
Change sentence

https://github.com/FreshRSS/FreshRSS/pull/1995#discussion_r214515954

Alkarex committed:
Merge branch 'FreshRSS/dev' into mark-as-unread

Alkarex committed:
Enable mark-as-unread only when unread articles are shown

In order to prevent erroneous marking-as-unread.
We might find a better logic later.

Alkarex commented:
I have disabled *mark-as-unread* when unread articles are not shown.
This is to prevent users from marking-as-unread the whole reader/category/feed too easily. It might be possible to find a better logic (the ones I have in mind has the drawback of always requiring an additional expensive SQL request at each view to find the boundaries, or require some not-so-nice JavaScript).

Alkarex committed:
Disable instead of hide mark-as-unread option

To make it easier to discover

Alkarex commented:
![image](https://user-images.githubusercontent.com/1008324/44998991-53d44c00-afba-11e8-862c-fbbe4dced152.png)

Alkarex commented:
@Frenzie Ready for translation PR ;-)

Alkarex commented on app/Controllers/entryController.php:
```diff
@@ -63,39 +64,38 @@ public function readAction() {
 			if (!$get) {
 				// No get? Mark all entries as read (from $id_max)
-				$entryDAO->markReadEntries($id_max);
+				$entryDAO->markReadEntries($id_max, $is_read);
```
Fixed in https://github.com/FreshRSS/FreshRSS/pull/4456
